modules/@apostrophecms/schema/ui/apos/logic/AposInputSlug.js

```diff
@@ -66,29 +65,43 @@ export default {
         delete newClone.archived;
         delete oldClone.archived;
 
-        oldValue = Object.values(oldClone).join(' ');
-        oldValue = oldValue.replace(/\//g, ' ');
+        // TODO: Do we really need to rely on all following values?
+        const oldVal = Object
+          .values(oldClone)
+          .join(' ')
+          .replace(/\//g, ' ');
 
-        newValue = Object.values(newClone).join(' ');
-        newValue = newValue.replace(/\//g, ' ');
+        const value = Object
+          .values(newClone)
+          .join(' ')
+          .replace(/\//g, ' ');
 
-        if (this.compatible(oldValue, this.next) && !newValue.archived) {
-          // If this is a page slug, we only replace the last section of the slug.
-          if (this.field.page) {
-            let parts = this.next.split('/');
-            parts = parts.filter(part => part.length > 0);
-            if ((!this.originalSlugPartsLength && parts.length) || (this.originalSlugPartsLength && parts.length === (this.originalSlugPartsLength - 1))) {
-              // Remove last path component so we can replace it
-              parts.pop();
-            }
-            parts.push(this.slugify(newValue, { componentOnly: true }));
-            if (parts[0].length) {
-              // TODO: handle page archives.
-              this.next = `/${parts.join('/')}`;
-            }
-          } else {
-            this.next = this.slugify(newValue);
-          }
+        if (oldVal === value) {
+          return;
+        }
```

Comment:
We don't want to run the code if the `followingValues` didn't change since it can create infinite recursion between this watcher and slugify.